### DISCUSSION TITLE
Inline media

### DISCUSSION THREAD

**Discussion by ghost** [GitSquared/edex-ui#1154] [Ideas]
Here's a FAQ on inline graphics at the command line:
https://twitter.com/thingskatedid/status/1316074032379248640
[Kitty](https://sw.kovidgoyal.net/kitty/graphics-protocol/), [libsixel](https://saitoha.github.io/libsixel/), and [iTerm2](https://iterm2.com/documentation-images.html) all have protocols for inline graphics at a terminal.  Would be cool to support one of them or to implement your own.  

At the moment, you've got an experimental globe feature that parses IP addresses out of the console result; iTerm2 does something similar.  Quoting their docs:

> iTerm2 extends the xterm protocol with a set of proprietary escape sequences. In general, the pattern is:
>     ESC ] 1337 ; key = value ^G
> Whitespace is shown here for ease of reading: in practice, no spaces should be used.
> For file transfer and inline images, the code is:
>     ESC ] 1337 ; File = [arguments] : base-64 encoded file contents ^G

You could do something similar but pass a JSON object between `ESC ] 1337 ;` and `^G`.

**Comment by GitSquared** [comment]
This is out of scope for edex, it should be implemented in `xterm.js`

**Reply by ghost** [reply]
Thx.  For future reference, here's their bug for supporting inline images:
https://github.com/xtermjs/xterm.js/pull/2503

That said, xterm.js does expose the ability to register handlers for [CSI, DCS, ESC, and OSC sequences](https://xtermjs.org/docs/guides/hooks/). iTerm2 uses an OSC sequence. edex-ui could similarly claim any unused OSC number and register a handler for it.

**Reply by ghost** [reply]
Here's a list of [widely supported OSC sequences to avoid](https://docs.google.com/spreadsheets/d/19W-lXWS9jYwqCK-LwgYo31GucPPxYVld_hVEcfpNpXg/edit#gid=433919454).
